fastlane/lib/fastlane/plugins/plugin_manager.rb

```diff
@@ -0,0 +1,338 @@
+      # state we'd already built up during this run. That was causing the command
+      # to fail, telling us to install the Gem we'd just introduced, even though
+      # that is exactly what we are trying to do!
+      #
+      # Bundler.with_clean_env solves this problem by resetting Bundler state before the
+      # exec'd call gets merged into this process.
+
+      Bundler.with_clean_env do
+        yield if block_given?
+      end
+    end
+
+    #####################################################
+    # @!group Initial setup
+    #####################################################
+
+    def setup
+      UI.important("It looks like fastlane plugins are not yet set up for this project.")
+
+      path_to_gemfile = gemfile_path || DEFAULT_GEMFILE_PATH
+
+      if gemfile_content.to_s.length > 0
+        UI.important("fastlane will modify your existing Gemfile at path '#{path_to_gemfile}'")
+      else
+        UI.important("fastlane will create a new Gemfile at path '#{path_to_gemfile}'")
+      end
+
+      UI.important("This change is neccessary for fastlane plugins to work")
+
+      unless UI.confirm("Should fastlane modify the Gemfile at path '#{path_to_gemfile}' for you?")
+        UI.important("Please add the following code to '#{path_to_gemfile}':")
+        puts ""
+        puts self.class.code_to_attach.magenta # we use `puts` instead of `UI` to make it easier to copy and paste
+        UI.user_error!("Please update '#{path_to_gemfile} and run fastlane again")
+      end
+
+      attach_plugins_to_gemfile!(path_to_gemfile)
+      UI.success("Successfully modified '#{path_to_gemfile}'")
+    end
+
+    # The code required to load the Plugins file
+    def self.code_to_attach
+      if FastlaneFolder.path
+        fastlane_folder_name = File.basename(FastlaneFolder.path)
+      else
+        fastlane_folder_name = "fastlane"
+      end
+      "plugins_path = File.join(File.dirname(__FILE__), '#{fastlane_folder_name}', '#{PluginManager::PLUGINFILE_NAME}')\n" \
+      "eval(File.read(plugins_path), binding) if File.exist?(plugins_path)"
+    end
+
+    # Makes sure, the user's Gemfile actually loads the Plugins file
+    def plugins_attached?
+      gemfile_path && gemfile_content.include?(self.class.code_to_attach)
+    end
+
+    def ensure_plugins_attached!
+      return if plugins_attached?
+      self.setup
+    end
+
+    #####################################################
+    # @!group Requiring the plugins
+    #####################################################
+
+    # Iterate over all available plugins
+    # which follow the naming convention
+    #   fastlane-plugin-[plugin_name]
+    # This will make sure to load the action
+    # and all its helpers
+    def self.load_plugins
+      UI.verbose("Checking if there are any plugins that should be loaded...")
+
+      loaded_plugins = false
+      Gem::Specification.each do |current_gem|
```

Comment:
I believe that this will not find plugins which are specified in Bundler via `git:` 

http://bundler.io/git.html

> Because RubyGems lacks the ability to handle gems from git, any gems installed from a git repository will not show up in `gem list`.